### PR TITLE
using settings for deps is apparently enough in 2.11

### LINKED ISSUE #961: "customDeps" do not resolve in Spark 2.3.0
Using the customDeps argument in the notebook to import libraries with sbt has been a great feature but breaks with 2.3.0.
For instance a notebook with metadata
```
"customDeps": [
     "mysql % mysql-connector-java % 8.0.11"
  ]
```
will allow us to do an import like
```
import com.mysql.jdbc.Driver
```
Now we have to manually place dependencies in the /lib directory or modify the build.sbt to add them at compile.

### PR DESCRIPTION
**Pull request by andypetrella**
fixes #961 

cc @GabeChurch, @vidma, @frbayart

### CODE REVIEW

File: modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala [hunk 1]
```diff
@@ -103,11 +103,13 @@ class Repl(val compilerOpts: List[String], val jars:List[String]=Nil) extends Re
...
 
-    jars.foreach { jar =>
+    val fps = jars.map{ jar =>
       import scala.tools.nsc.util.ClassPath
       val f = scala.tools.nsc.io.File(jar).normalize
-      loop.addedClasspath = ClassPath.join(loop.addedClasspath, f.path)
+      //loop.addedClasspath = ClassPath.join(loop.addedClasspath, f.path)
+      f.path
     }
+    settings.classpath.value=fps.mkString(java.io.File.pathSeparator)
```
Comment (vidma) [inline modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala]: is it OK that this is overriding and loosing some older values of `settings.classpath.value`?
see https://github.com/spark-notebook/spark-notebook/pull/964/files#diff-42e4d431da236e133a4c232a417ac431R94

Comment (andypetrella) [reply modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala]: Good point!!
I guess we could do : 
`settings.classpath.value=(classpath.distinct ::: fps).mkString(java.io.File.pathSeparator)`

and remove that other line?

Comment (vidma) [reply modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala]: sounds legit

File: modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala [hunk 2]
```diff
@@ -103,11 +102,13 @@ class Repl(val compilerOpts: List[String], val jars:List[String]=Nil) extends Re
...
 
-    jars.foreach { jar =>
+    val fps = jars.map{ jar =>
       import scala.tools.nsc.util.ClassPath
       val f = scala.tools.nsc.io.File(jar).normalize
-      loop.addedClasspath = ClassPath.join(loop.addedClasspath, f.path)
+      //loop.addedClasspath = ClassPath.join(loop.addedClasspath, f.path)
+      f.path
     }
+    settings.classpath.value=(classpath.distinct ::: fps).mkString(java.io.File.pathSeparator)
```
Comment (vidma) [inline modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala]: I wonder if this sets driver classpath or both driver & executor?

Comment (andypetrella) [reply modules/spark/src/main/scala_2.11/spark-last/notebook/kernel/Repl.scala]: only driver, executor is done through the spark.jars conf

**Pull request merged**